Root tests involving non-existing URLs fail (at my locale)

I get testing failures at my locale of this type :

```
sphinx$ make test TEST=tests/test_build_linkcheck.py::test_defaults
============================= test session starts ==============================
platform darwin -- Python 3.6.8, pytest-4.1.1, py-1.7.0, pluggy-0.8.1 -- /Users/johndoe/Downloads/sphinxvenv/bin/python3
cachedir: .pytest_cache
libraries: Sphinx-2.1.1+/3360f2461, docutils-0.14
base tempdir: /var/folders/np/4_9pppfs54xdv9m79m7rwx4w0000gn/T/pytest-of-johndoe/pytest-5
rootdir: /Users/johndoe/_gits/sphinx, inifile: setup.cfg
collected 1 item                                                               

tests/test_build_linkcheck.py::test_defaults FAILED                      [100%]

=================================== FAILURES ===================================
________________________________ test_defaults _________________________________

app = <SphinxTestApp buildername='linkcheck'>
status = <_io.StringIO object at 0x10f3d6ca8>
warning = <_io.StringIO object at 0x10f3d6798>

    @pytest.mark.sphinx('linkcheck', testroot='linkcheck', freshenv=True)
    def test_defaults(app, status, warning):
        app.builder.build_all()
    
        assert (app.outdir / 'output.txt').exists()
        content = (app.outdir / 'output.txt').text()
    
        print(content)
        # looking for '#top' and 'does-not-exist' not found should fail
        assert "Anchor 'top' not found" in content
        assert "Anchor 'does-not-exist' not found" in content
        # looking for non-existent URL should fail
        assert " Max retries exceeded with url: /doesnotexist" in content
        # images should fail
>       assert "Not Found for url: http://example.com/image.png" in content
E       assert 'Not Found for url: http://example.com/image.png' in "links.txt:13: [broken] https://localhost:7777/doesnotexist: HTTPSConnectionPool(host='localhost', port=7777): Max ret...nection.HTTPConnection object at 0x1115850b8>: Failed to establish a new connection: [Errno 65] No route to host',))\n"

tests/test_build_linkcheck.py:28: AssertionError
----------------------------- Captured stdout call -----------------------------
links.txt:13: [broken] https://localhost:7777/doesnotexist: HTTPSConnectionPool(host='localhost', port=7777): Max retries exceeded with url: /doesnotexist (Caused by NewConnectionError('<urllib3.connection.VerifiedHTTPSConnection object at 0x11145ccc0>: Failed to establish a new connection: [Errno 61] Connection refused',))
links.txt:11: [broken] http://example.com/#top: Anchor 'top' not found
links.txt:16: [broken] http://example.com/image2.png: 404 Client Error: Not Found for url: http://example.com/image2.png
links.txt:12: [broken] http://www.sphinx-doc.org/en/1.7/intro.html#does-not-exist: Anchor 'does-not-exist' not found
links.txt:9: [broken] http://example.com/#!bar: HTTPConnectionPool(host='example.com', port=80): Max retries exceeded with url: / (Caused by NewConnectionError('<urllib3.connection.HTTPConnection object at 0x11145c278>: Failed to establish a new connection: [Errno 65] No route to host',))
links.txt:15: [broken] http://example.com/image.png: HTTPConnectionPool(host='example.com', port=80): Max retries exceeded with url: /image.png (Caused by NewConnectionError('<urllib3.connection.HTTPConnection object at 0x1115850b8>: Failed to establish a new connection: [Errno 65] No route to host',))

--------------------------- Captured stdout teardown ---------------------------
# testroot: root
# builder: linkcheck
# srcdir: /private/var/folders/np/4_9pppfs54xdv9m79m7rwx4w0000gn/T/pytest-of-johndoe/pytest-5/linkcheck
# outdir: /private/var/folders/np/4_9pppfs54xdv9m79m7rwx4w0000gn/T/pytest-of-johndoe/pytest-5/linkcheck/_build/linkcheck
# status: 
Running Sphinx v2.1.1+/3360f2461
building [linkcheck]: all source files
updating environment: 1 added, 0 changed, 0 removed
reading sources... [100%] links                                                
looking for now-outdated files... none found
pickling environment... done
checking consistency... done
preparing documents... done
writing output... [100%] links                                                 
(line   10) ok        http://example.com#!bar
(line   13) broken    https://localhost:7777/doesnotexist - HTTPSConnectionPool(host='localhost', port=7777): Max retries exceeded with url: /doesnotexist (Caused by NewConnectionError('<urllib3.connection.VerifiedHTTPSConnection object at 0x11145ccc0>: Failed to establish a new connection: [Errno 61] Connection refused',))
(line   11) broken    http://example.com/#top - Anchor 'top' not found
(line    3) ok        https://www.w3.org/TR/2006/REC-xml-names-20060816/#defaulting
(line   16) broken    http://example.com/image2.png - 404 Client Error: Not Found for url: http://example.com/image2.png
(line   12) broken    http://www.sphinx-doc.org/en/1.7/intro.html#does-not-exist - Anchor 'does-not-exist' not found
(line    9) broken    http://example.com/#!bar - HTTPConnectionPool(host='example.com', port=80): Max retries exceeded with url: / (Caused by NewConnectionError('<urllib3.connection.HTTPConnection object at 0x11145c278>: Failed to establish a new connection: [Errno 65] No route to host',))
(line   15) broken    http://example.com/image.png - HTTPConnectionPool(host='example.com', port=80): Max retries exceeded with url: /image.png (Caused by NewConnectionError('<urllib3.connection.HTTPConnection object at 0x1115850b8>: Failed to establish a new connection: [Errno 65] No route to host',))


# warning: 
/private/var/folders/np/4_9pppfs54xdv9m79m7rwx4w0000gn/T/pytest-of-johndoe/pytest-5/linkcheck/links.txt:9: WARNING: Duplicate explicit target name: "example bar invalid".
/private/var/folders/np/4_9pppfs54xdv9m79m7rwx4w0000gn/T/pytest-of-johndoe/pytest-5/linkcheck/links.txt:9: WARNING: Duplicate explicit target name: "example bar invalid".
/private/var/folders/np/4_9pppfs54xdv9m79m7rwx4w0000gn/T/pytest-of-johndoe/pytest-5/linkcheck/links.txt:5: WARNING: Unknown target name: "http://www.sphinx-doc.org/en/1.4.8/tutorial.html#install-sphinx".

========================== 1 failed in 75.99 seconds ===========================
make: *** [Makefile:72: test] Error 1
```

Notice the long wait of 76 seconds.

Two other tests for same non-existing URLs reasons:

```
tests/test_build_latex.py::test_latex_images FAILED                      [ 50%]
...
tests/test_build_linkcheck.py::test_defaults FAILED                      [ 51%]
tests/test_build_linkcheck.py::test_anchors_ignored FAILED               [ 51%]
...
```

- Mac OS X 10.9.5
- Python 3.6.8
- Sphinx 2.1.0

Surprisingly I also had earlier a test where test_anchors_ignored did not fail

```
tests/test_build_latex.py::test_latex_images FAILED                      [ 50%]
...
tests/test_build_linkcheck.py::test_defaults FAILED                      [ 51%]
tests/test_build_linkcheck.py::test_anchors_ignored PASSED               [ 51%]
...
======== 2 failed, 1270 passed, 2 skipped, 7 warnings in 298.55 seconds ========
make: *** [Makefile:72: test] Error 1
```

but I got on my third try again 3 tests failing, at same commit 2.1.1/3360f2461

```
======== 3 failed, 1269 passed, 2 skipped, 7 warnings in 371.21 seconds ========
make: *** [Makefile:72: test] Error 1
```

I have the same problem with ``tests/test_build_linkcheck.py::test_defaults`` at my locale on a checkout of v1.7.6.

As I had no such failures in past, could it be due to some change in `urllib3` error messages?

```
$ make test TEST=tests/test_build_linkcheck.py::test_defaults
Temporary files will be placed in /Users/johndoe/_gits/sphinx/tests/build.
============================= test session starts ==============================
platform darwin -- Python 3.6.8, pytest-4.1.1, py-1.7.0, pluggy-0.8.1 -- /Users/johndoe/sphinxvenv/bin/python
cachedir: .pytest_cache
Running Sphinx test suite (with Python 3.6.8)...
rootdir: /Users/johndoe/_gits/sphinx, inifile: setup.cfg
collected 1 item                                                               

tests/test_build_linkcheck.py::test_defaults FAILED                      [100%]

=================================== FAILURES ===================================
________________________________ test_defaults _________________________________

app = <SphinxTestApp buildername='linkcheck'>
status = <_io.StringIO object at 0x1076a9dc8>
warning = <_io.StringIO object at 0x1076dc5e8>

    @pytest.mark.sphinx('linkcheck', testroot='linkcheck', freshenv=True)
    def test_defaults(app, status, warning):
        app.builder.build_all()
    
        assert (app.outdir / 'output.txt').exists()
        content = (app.outdir / 'output.txt').text()
    
        print(content)
        # looking for #top should fail
        assert "Anchor 'top' not found" in content
>       assert len(content.splitlines()) == 1
E       assert 3 == 1
E         -3
E         +1

tests/test_build_linkcheck.py:26: AssertionError
----------------------------- Captured stdout call -----------------------------
links.txt:11: [broken] http://example.com/#top: Anchor 'top' not found
links.txt:9: [broken] http://example.com/#!bar: HTTPConnectionPool(host='example.com', port=80): Max retries exceeded with url: / (Caused by NewConnectionError('<urllib3.connection.HTTPConnection object at 0x107be22e8>: Failed to establish a new connection: [Errno 65] No route to host',))
links.txt:10: [broken] http://example.com#!bar: HTTPConnectionPool(host='example.com', port=80): Max retries exceeded with url: / (Caused by NewConnectionError('<urllib3.connection.HTTPConnection object at 0x107d88c18>: Failed to establish a new connection: [Errno 65] No route to host',))

--------------------------- Captured stdout teardown ---------------------------
# testroot: root
# builder: linkcheck
# srcdir: /Users/johndoe/_gits/sphinx/tests/build/linkcheck
# outdir: /Users/johndoe/_gits/sphinx/tests/build/linkcheck/_build/linkcheck
# status: 
Running Sphinx v1.7.6
building [linkcheck]: all source files
updating environment: 1 added, 0 changed, 0 removed
reading sources... [100%] links                                                
looking for now-outdated files... none found
pickling environment... done
checking consistency... done
preparing documents... done
writing output... [100%] links                                                 
(line   11) broken    http://example.com/#top - Anchor 'top' not found
(line    3) ok        https://www.w3.org/TR/2006/REC-xml-names-20060816/#defaulting
(line    9) broken    http://example.com/#!bar - HTTPConnectionPool(host='example.com', port=80): Max retries exceeded with url: / (Caused by NewConnectionError('<urllib3.connection.HTTPConnection object at 0x107be22e8>: Failed to establish a new connection: [Errno 65] No route to host',))
(line   10) broken    http://example.com#!bar - HTTPConnectionPool(host='example.com', port=80): Max retries exceeded with url: / (Caused by NewConnectionError('<urllib3.connection.HTTPConnection object at 0x107d88c18>: Failed to establish a new connection: [Errno 65] No route to host',))


# warning: 
/Users/johndoe/_gits/sphinx/tests/build/linkcheck/links.txt:9: WARNING: Duplicate explicit target name: "example bar invalid".
/Users/johndoe/_gits/sphinx/tests/build/linkcheck/links.txt:9: WARNING: Duplicate explicit target name: "example bar invalid".
/Users/johndoe/_gits/sphinx/tests/build/linkcheck/links.txt:5: WARNING: Unknown target name: "http://www.sphinx-doc.org/en/1.4.8/tutorial.html#install-sphinx".

========================== 1 failed in 75.72 seconds ===========================
make: *** [Makefile:72: test] Error 1
```

--- a/tests/roots/test-images/index.rst
+++ b/tests/roots/test-images/index.rst
@@ -26,4 +26,4 @@ test-image
 .. image:: https://www.python.org/static/img/python-logo.png
 
 .. non-exist remote image
-.. image:: http://example.com/NOT_EXIST.PNG
+.. image:: https://www.google.com/NOT_EXIST.PNG

--- a/tests/roots/test-intl/external_links.txt
+++ b/tests/roots/test-intl/external_links.txt
@@ -23,8 +23,8 @@ link to external1_ and external2_.
 
 link to `Sphinx Site <http://sphinx-doc.org>`_ and `Python Site <http://python.org>`_.
 
-.. _external1: http://example.com/external1
-.. _external2: http://example.com/external2
+.. _external1: https://www.google.com/external1
+.. _external2: https://www.google.com/external2
 
 
 Multiple references in the same line

--- a/tests/roots/test-linkcheck/links.txt
+++ b/tests/roots/test-linkcheck/links.txt
@@ -6,11 +6,11 @@ This is from CPython documentation.
 
 Some additional anchors to exercise ignore code
 
-* `Example Bar invalid <http://example.com/#!bar>`_
-* `Example Bar invalid <http://example.com#!bar>`_ tests that default ignore anchor of #! does not need to be prefixed with /
-* `Example Bar invalid <http://example.com/#top>`_
+* `Example Bar invalid <https://www.google.com/#!bar>`_
+* `Example Bar invalid <https://www.google.com#!bar>`_ tests that default ignore anchor of #! does not need to be prefixed with /
+* `Example Bar invalid <https://www.google.com/#top>`_
 * `Example anchor invalid <http://www.sphinx-doc.org/en/1.7/intro.html#does-not-exist>`_
 * `Complete nonsense <https://localhost:7777/doesnotexist>`_
 
-.. image:: http://example.com/image.png
-.. figure:: http://example.com/image2.png
+.. image:: https://www.google.com/image.png
+.. figure:: https://www.google.com/image2.png

--- a/tests/test_build_html.py
+++ b/tests/test_build_html.py
@@ -1239,25 +1239,25 @@ def test_html_entity(app):
 def test_html_inventory(app):
     app.builder.build_all()
     with open(app.outdir / 'objects.inv', 'rb') as f:
-        invdata = InventoryFile.load(f, 'http://example.com', os.path.join)
+        invdata = InventoryFile.load(f, 'https://www.google.com', os.path.join)
     assert set(invdata.keys()) == {'std:label', 'std:doc'}
     assert set(invdata['std:label'].keys()) == {'modindex', 'genindex', 'search'}
     assert invdata['std:label']['modindex'] == ('Python',
                                                 '',
-                                                'http://example.com/py-modindex.html',
+                                                'https://www.google.com/py-modindex.html',
                                                 'Module Index')
     assert invdata['std:label']['genindex'] == ('Python',
                                                 '',
-                                                'http://example.com/genindex.html',
+                                                'https://www.google.com/genindex.html',
                                                 'Index')
     assert invdata['std:label']['search'] == ('Python',
                                               '',
-                                              'http://example.com/search.html',
+                                              'https://www.google.com/search.html',
                                               'Search Page')
     assert set(invdata['std:doc'].keys()) == {'index'}
     assert invdata['std:doc']['index'] == ('Python',
                                            '',
-                                           'http://example.com/index.html',
+                                           'https://www.google.com/index.html',
                                            'The basic Sphinx documentation for testing')
 
 

--- a/tests/test_build_latex.py
+++ b/tests/test_build_latex.py
@@ -1242,7 +1242,7 @@ def test_latex_images(app, status, warning):
     # not found images
     assert '\\sphinxincludegraphics{{NOT_EXIST}.PNG}' not in result
     assert ('WARNING: Could not fetch remote image: '
-            'http://example.com/NOT_EXIST.PNG [404]' in warning.getvalue())
+            'https://www.google.com/NOT_EXIST.PNG [404]' in warning.getvalue())
 
     # an image having target
     assert ('\\sphinxhref{https://www.sphinx-doc.org/}'

--- a/tests/test_build_linkcheck.py
+++ b/tests/test_build_linkcheck.py
@@ -25,8 +25,8 @@ def test_defaults(app, status, warning):
     # looking for non-existent URL should fail
     assert " Max retries exceeded with url: /doesnotexist" in content
     # images should fail
-    assert "Not Found for url: http://example.com/image.png" in content
-    assert "Not Found for url: http://example.com/image2.png" in content
+    assert "Not Found for url: https://www.google.com/image.png" in content
+    assert "Not Found for url: https://www.google.com/image2.png" in content
     assert len(content.splitlines()) == 5
 
 
@@ -36,8 +36,8 @@ def test_defaults(app, status, warning):
                    'linkcheck_ignore': [
                        'https://localhost:7777/doesnotexist',
                        'http://www.sphinx-doc.org/en/1.7/intro.html#',
-                       'http://example.com/image.png',
-                       'http://example.com/image2.png']
+                       'https://www.google.com/image.png',
+                       'https://www.google.com/image2.png']
                    })
 def test_anchors_ignored(app, status, warning):
     app.builder.build_all()

--- a/tests/test_intl.py
+++ b/tests/test_intl.py
@@ -885,8 +885,8 @@ def test_xml_keep_external_links(app):
     assert_elem(
         para1[0],
         ['LINK TO', 'external2', 'AND', 'external1', '.'],
-        ['http://example.com/external2',
-         'http://example.com/external1'])
+        ['https://www.google.com/external2',
+         'https://www.google.com/external1'])
     assert_elem(
         para1[1],
         ['LINK TO', 'THE PYTHON SITE', 'AND', 'THE SPHINX SITE', '.'],

--- a/tests/test_markup.py
+++ b/tests/test_markup.py
@@ -279,9 +279,9 @@ def get_verifier(verify, verify_re):
     (
         # in URIs
         'verify_re',
-        '`test <http://example.com/~me/>`_',
+        '`test <https://www.google.com/~me/>`_',
         None,
-        r'\\sphinxhref{http://example.com/~me/}{test}.*',
+        r'\\sphinxhref{https://www.google.com/~me/}{test}.*',
     ),
     (
         # description list: simple

